packages/@aws-cdk/aws-codepipeline-api/lib/action.ts

```diff
@@ -270,16 +222,46 @@ export abstract class Action extends cdk.Construct {
     );
   }
 
-  protected addOutputArtifact(name: string = this.stage._internal._generateOutputArtifactName(this)): Artifact {
-    const artifact = new Artifact(this, name);
+  protected addOutputArtifact(name: string): Artifact {
+    const artifact = new Artifact(name);
     this._actionOutputArtifacts.push(artifact);
     return artifact;
   }
 
-  protected addInputArtifact(artifact: Artifact = this.stage._internal._findInputArtifact(this)): Action {
+  protected addInputArtifact(artifact: Artifact): Action {
     this._actionInputArtifacts.push(artifact);
     return this;
   }
+
+  protected getParent(methodName: string): cdk.Construct {
```

Comment:
As this is a customer-facing error, are you OK with actually leaving this? I'd rather not rely on only the stack trace to communicate this issue, as it can be quite surprising.

Reply:
Stack traces are part of the error. There is no need to duplicate this information and complicate this __public__ API.

Reply:
Done.